spec/address_book_spec.rb

```diff
@@ -109,4 +109,91 @@ def check_entry(entry, expected_name, expected_number, expected_email)
 
   end
 
+
+  context "#binary_search" do
+    it "searches AddressBook for a non-existent entry" do
+      book.import_from_csv("entries.csv")
+      entry = book.binary_search("Dan")
+      expect(entry).to be_nil
+    end
+  end
+
+  it "searches AddressBook for Bill" do
+    book.import_from_csv("entries.csv")
+    entry = book.binary_search("Bill")
+    expect entry.instance_of?(Entry)
+    check_entry(entry, "Bill", "555-555-4854", "bill@blocmail.com")
+  end
+
+  it "searches AddressBook for Bob" do
+    book.import_from_csv("entries.csv")
+    entry = book.binary_search("Bob")
+    expect entry.instance_of?(Entry)
+    check_entry(entry, "Bob", "555-555-5415", "bob@blocmail.com")
+  end
+
+  it "searches AddressBook for Joe" do
+    book.import_from_csv("entries.csv")
+    entry = book.binary_search("Joe")
+    expect entry.instance_of?(Entry)
+    check_entry(entry, "Joe", "555-555-3660", "joe@blocmail.com")
+  end
+
+  it "searches AddressBook for Sally" do
+    book.import_from_csv("entries.csv")
+    entry = book.binary_search("Sally")
+    expect entry.instance_of?(Entry)
+    check_entry(entry, "Sally", "555-555-4646", "sally@blocmail.com")
+  end
+
+  it "searches AddressBook for Sussie" do
+    book.import_from_csv("entries.csv")
+    entry = book.binary_search("Sussie")
+    expect entry.instance_of?(Entry)
+    check_entry(entry, "Sussie", "555-555-2036", "sussie@blocmail.com")
+  end
+
+  context "#iterative_search" do
```

Comment:
Similiarly, all of the `iterative_search` `it` blocks would be inside the `#iterative_search context` block.